pkg/cmd/app.go

```diff
@@ -228,6 +236,19 @@ restart:
 		return err
 	}
 
+	if config.Debug {
+		spew.Config.Indent = " "
+		spew.Config.DisablePointerAddresses = true
+		spew.Config.DisableCapacities = true
+		//enable debug logging
+		logrus.SetLevel(logrus.DebugLevel)
+		logrus.Debug("Debug output is enabled")
+	}
+
+	logrus.Debugf("Command line: %s", strings.Join(os.Args, " "))
+
+	logrus.WithField(dcgmexporter.LoggerDumpField, spew.Sdump(config)).Debug("Loaded configuration")
```

Comment:
One option could be to use `fmt.Sprintf("%+v", config)`.

It changes output from (includes type and len information):
```
dump="(*dcgmexporter.Config)({\n CollectorsFile: (string) (len=26) \"./etc/default-counters.csv\",\n Address: (string) (len=5) \":9400\",\n CollectInterval: (int) 30000,\n Kubernetes: (bool) false,\n KubernetesGPUIdType: (dcgmexporter.KubernetesGPUIDType) (len=3) \"uid\",\n CollectDCP: (bool) true,\n UseOldNamespace: (bool) false,\n UseRemoteHE: (bool) false,\n RemoteHEInfo: (string) (len=14) \"localhost:5555\",\n GPUDevices: (dcgmexporter.DeviceOptions) {\n  Flex: (bool) true,\n  MajorRange: ([]int) <nil>,\n  MinorRange: ([]int) <nil>\n },\n SwitchDevices: (dcgmexporter.DeviceOptions) {\n  Flex: (bool) true,\n  MajorRange: ([]int) <nil>,\n  MinorRange: ([]int) <nil>\n },\n CPUDevices: (dcgmexporter.DeviceOptions) {\n  Flex: (bool) true,\n  MajorRange: ([]int) <nil>,\n  MinorRange: ([]int) <nil>\n },\n NoHostname: (bool) false,\n UseFakeGPUs: (bool) false,\n ConfigMapData: (string) (len=4) \"none\",\n MetricGroups: ([]dcgm.MetricGroup) <nil>,\n WebSystemdSocket: (bool) false,\n WebConfigFile: (string) \"\",\n XIDCountWindowSize: (int) 300000,\n ReplaceBlanksInModelName: (bool) false,\n Debug: (bool) true\n})\n"
```

To (excludes type and len information)::
```
dump="&{CollectorsFile:./etc/default-counters.csv Address::9400 CollectInterval:30000 Kubernetes:false KubernetesGPUIdType:uid CollectDCP:true UseOldNamespace:false UseRemoteHE:false RemoteHEInfo:localhost:5555 GPUDevices:{Flex:true MajorRange:[] MinorRange:[]} SwitchDevices:{Flex:true MajorRange:[] MinorRange:[]} CPUDevices:{Flex:true MajorRange:[] MinorRange:[]} NoHostname:false UseFakeGPUs:false ConfigMapData:none MetricGroups:[] WebSystemdSocket:false WebConfigFile: XIDCountWindowSize:300000 ReplaceBlanksInModelName:false Debug:true}"
```

Reply:
done.